helm-charts/splunk-otel-collector/templates/daemonset.yaml

```diff
@@ -269,6 +269,8 @@ spec:
           {{- include "splunk-otel-collector.securityContext" (dict "isWindows" .Values.isWindows "securityContext" $agent.securityContext "setRunAsUser" true) | nindent 10 }}
         {{- end }}
         env:
+          - name: GOMEMLIMIT
+            value: {{ include "splunk-otel-collector.convertMemToMib" $agent.resources.limits.memory | int64 }}
           - name: SPLUNK_MEMORY_TOTAL_MIB
```

Comment:
Shouldn't `SPLUNK_MEMORY_TOTAL_MIB` be removed? If not how it interacts with `GOMEMLIMIT`?

Reply:
As far as I understand it has no effect from the tests on the internal realms. this is a good point, may be we can investigate the idea of removing it.

Reply:
If we can set it dynamically in https://github.com/signalfx/splunk-otel-collector/blob/91f039d015bdd5792e172ea635bc9b1dfd07a101/internal/settings/settings.go from `SPLUNK_MEMORY_TOTAL_MIB`, that would be great. So we don't need to provide two separate env vars. I remember @atoulme mentioned that he tried that.